OpenLineage/OpenLineage
# CI and deploy steps for docs

*Problem*: we want to have a single repo where developers push code and documentation changes within a single PR. 

As a result of the above, we moved `Openlineage/docs` repo into the main repository. In future we would like to allow versioning of the whole website but we don't want to store whole history in the main repo. Because this will get too big, because this should contain javadoc and we don't want to store javadoc within the main repo. 

As a solution to above, we decided to store  current version of the docs in the main repo while having a separate `OpenLineage-site` repo for the versioned documentation. Website will be served as GitHub page of `Openlineage-site` repo. In future, a separate GitHub action will create a new version of the site within `openlineage-site` while keeping the all the other versions. This will be done automatically. Users should only push changes to the main repo. Synchronisation with additional repo should be hidden.  This PR adds a logic to synchronise current state to additional repo with a javadoc and openapi page being generated and pushed into the repo. 

Moreover, documentation is also used to server facet and Openlineage spec definitions. Whenever, a version of a facet is bumped `1.0.5` -> `1.0.6`, a process should create a new copy of facet definition and store it under `/1.0.6/` path so that it is reachable via URL.  This is done in precommit and verified on CI site. Precommit generates a copy of facets changes and puts it under proper location. Additionally, a CI step verifies if the content is generated.

The PR in its current shape is pushing changes into my fork `pawel-big-lebowski/openlineage-site`. This is happening (bcz I couldn't obtain fine-grained personal access token to `OpenLineage/openlineage-site`. Anyway, this is good for testing. 

To verify a correctness of the code created: 
 * Visit [`pawel-big-lebowski.github.io/openlineage-site`](https://pawel-big-lebowski.github.io/openlineage-site/)
 * Verify openapi is generated  -> https://pawel-big-lebowski.github.io/openlineage-site/apidocs/openapi/
 * Verify javadoc is generated -> https://pawel-big-lebowski.github.io/openlineage-site/apidocs/javadoc/
 * Modify a spec (bump facet version or version in OpenLineage.json) to see recommit works as expected
 * See https://github.com/pawel-big-lebowski/OpenLineage-site to view last commits. For example this is an initial commit -> https://github.com/pawel-big-lebowski/openlineage-site/commit/d818a1333f4902d770d5d796ac76d9f6a7b98cc0
